proposal.rst

```diff
@@ -180,39 +180,57 @@ Motivations
+
+	* FIRE queries stored in inappropriate format, they should be transformed to use them in Xapian; It will possible to use English language corpus, because Eglish stemmer exists in Xapian.
+
+	* TREC dataset has big enough amount of documents, it is possible to combine some of them during pasing to create more complex queries.
+
+* Implement evaluator. According to xapian-letor/tests/api_letor.cc it is no big difference in applying rankers to data, it should just apply algorithms and return scores, so it could be another one class in xapian/letor. It would be nice to be able to call it same way as make check.
+
+* Write suite for combining documents. TREC dataset has big corpus of small documents. To get more content small documents could be merged. Also queries and qrels have to be transformed. After that evaluation could be done using Xapian API logic operations for queries. Since it is not needed for every Letor run, it should be separate tool, like xapian-letor/bin/xapian-prepare-trainingfile.cc. 
+
+* It is possible to use Wikipedia dumps as data for perftest. `Wikipedia <https://dumps.wikimedia.org/enwiki/latest/>`_ corpus just stores XML file with links to articles and some info, so it each document should be downloaded and indexed individually. Also plain text `extractor <https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/corpora/wikicorpus.py>`_ and query fabricator needed.
 
 **Do you have any preliminary findings or results which suggest that your
 approach is possible and likely to succeed?**
 
-I worked with Xapian test harness, it has BackendManager class wich makes testing databases api easier. I believe ranking benchmark should be done in similar way. 
+I worked with Xapian test harness in `#758 <https://trac.xapian.org/ticket/758>`_, it has BackendManager class wich makes testing databases api easier. I believe ranking benchmark should be done in similar way.
 
 **What other approaches have you considered, and why did you reject those in
 favour of your chosen approach?**
 
-Another way is to create ranking suite as part of Letor, but it is not clear enough how to develop performance measurement api in this case.
+I have discussed the possibility of using web datasets like INEX, but there are several reasons to store dataset locally:
+
+* Most of Web-API datasets requires an account creation, which is not suitable for Xapian users.
+
+* All data should be stored locally at least in parts to be indexed.
 
 **Please note any uncertainties or aspects which depend on further research or
 investigation.**
 
-Databases could be too big to store them on local machine, or too big for memory, which could cause inaccuracies in performance tests, because of reading from disk. Also I don't know how different ranking algorithms are, so it could impact timeline.
+Databases indexing time could be big enouth, in connection with this, difficulties may arise during implementation.
```

Comment:
Big enough for what? What difficulties are you worried about? (There are always difficulties in implementation, but if you have specific ones you're concerned about, make that clear.)